src/main/java/org/idpass/lite/IdentFieldsConstraint.java

```diff
@@ -0,0 +1,132 @@
+package org.idpass.lite;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+
+/**
+ * The IdentFieldConstraint class is the input class
+ * to the Ident class. The getter methods to each field
+ * of interest should return the type expected by Ident
+ * class.
+ *
+ * The list of member fields in this class are the fields
+ * of interest to look for.
+ */
+
+public class IdentFieldsConstraint {
+
+    private String fullName;
+    private String surName;
+    private String givenName;
+    private Number UIN;
+    private String gender;
+    private String placeOfBirth;
+    private LocalDate dateOfBirth;
+    private String addressLine1;
+    private String addressLine2;
+    private String addressLine3;
+    private String region;
+    private String province;
+    private String postalCode;
+
+    /**
+     * The only constructor that accepts a map of found key/value pairs
+     * when the input json is traversed and using this class to provide the
+     * fields of interest.
+     * @param m Is a map of found key/value pairs from an input json
+     */
+
+    public IdentFieldsConstraint(Map<String, Object> m)
+    {
+        for (Map.Entry<String, Object> melem : m.entrySet()) {
+
+            String k = melem.getKey();
+
+            try {
+                Field f = IdentFieldsConstraint.class.getDeclaredField(k);
+
+                if (f.getType().isAssignableFrom(LocalDate.class)) {
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/d"); /// TODO: move to config? or list of possible combinations
+                    LocalDate dob = LocalDate.parse(melem.getValue().toString(), formatter);
+                    f.set(this, dob);
+                } else if(f.getType().isAssignableFrom(Number.class)) {
+                    Number num = (Number)melem.getValue();
+                    f.set(this, num);
+                } else{
+                    f.set(this, melem.getValue());
+                }
+
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+
```

Comment:
Should we print detail exception with field name and value here for error trace?

Reply:
Ok. I am adding more code in this part because I am also adding the expected type of the field. For example, `{"fullName":"java.lang.String", "dateOfBirth":"java.time.LocalDate"}`. Such that, if the field value does not render according to its expected type, then it is an error.